### PR TITLE
feat: impl Debug for all public structs

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

Fix https://github.com/rustls/rustls/issues/1052

### CODE REVIEW

**Comment by Xuanwo** [comment]
Cc @djc for reviewing, thank!

**Comment by djc** [comment]
I think this is adding too many `Debug` implementations. I think we should have `Debug` implementations only on public types and public fields. Any private types and private fields should not appear in a `Debug` impl.

**Comment by Xuanwo** [comment]
Cc @djc, I have removed all Debug for internal types. PTAL, thanks.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
If you rebase this on top of main, we'll get a working CI run.

**Comment by Xuanwo** [comment]
> If you rebase this on top of main, we'll get a working CI run.

Done!

File: rustls/src/lib.rs [hunk 1]
```diff
@@ -266,6 +266,7 @@
     trivial_numeric_casts,
     missing_docs,
     //unreachable_pub, https://github.com/rust-lang/rust/issues/102352
+    unreachable_pub,
```
Comment (djc) [inline rustls/src/lib.rs:269]: You rebased incorrectly.

Comment (Xuanwo) [reply rustls/src/lib.rs:269]: Oh, sorry.

**Review by djc — APPROVED** [review]

**Pull request merged**